pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -407,26 +426,28 @@ func createTracer(
 
 // Function to generate an event used most of the times.
 // Returns pid of executed process.
-func generateEvent() (int, error) {
+func generateEvent() (int, int, error) {
 	cmd := exec.Command("/bin/cat", "/dev/null")
 	if err := cmd.Run(); err != nil {
-		return 0, fmt.Errorf("running command: %w", err)
+		return 0, 0, fmt.Errorf("running command: %w", err)
 	}
 
-	return cmd.Process.Pid, nil
+	ptid := syscall.Gettid()
+
+	return cmd.Process.Pid, ptid, nil
 }
 
 // Function to generate a failed event.
 // Return 0 as no process is created.
-func generateFailedEvent() (int, error) {
+func generateFailedEvent() (int, int, error) {
 	// Ignore error since we want to capture a failed event
 	exec.Command("/bin/foobar").Run()
-	return 0, nil
+	return 0, 0, nil
 }
 
 // Function to generate an exec() event from a thread.
-func generateEventFromThread(success bool) func() (int, error) {
-	return func() (int, error) {
+func generateEventFromThread(success bool) func() (int, int, error) {
+	return func() (int, int, error) {
```

Comment:
Functions
- generateEvent()
- generateFailedEvent()
- generateEventFromThread()

Could return your new struct ExecResult:

```suggestion
func generateEventFromThread(success bool) func() (ExecResult, error) {
	return func() (ExecResult, error) {
```

Reply:
DONE